tests/system/test_ecs_mappings.py

```diff
@@ -0,0 +1,127 @@
+        Test that all fields are aliased or otherwise accounted for in ECS migration.
+        """
+        all_fields = set()
+        alias_source_fields = set()
+        alias_target_fields = set()
+        for f, a in flatmap(yaml.load(self.command_output)["mappings"]["doc"]["properties"]):
+            all_fields.add(f)
+            if a.get("type") == "alias":
+                alias_source_fields.add(f)
+                alias_target_fields.add(a["path"])
+
+        # fields with special exception, due to mapping type changes, etc
+        # no comment means unchanged
+        exception_fields = {
+            "@timestamp",
+            "container.labels",  # target for docker.container.labels copy
+            "context.request.url.port",  # field copy to url.port, keyword -> int
+            "context.request.url.protocol",  # field copy to url.scheme, drop trailing ":"
+            "context.tags",  # field copy, can't alias objects
+            "docker.container.labels",  # field copy, can't alias objects
+            "error.code", "error.culprit", "error.exception.code", "error.exception.handled", "error.exception.message",
+            "error.exception.module", "error.exception.type", "error.grouping_key", "error.id", "error.log.level",
+            "error.log.logger_name", "error.log.message", "error.log.param_message", "error.message", "error.type",
+            "fields",
+            "labels",  # target for context.tags copy
+            "listening",
+            "kubernetes.annotations", "kubernetes.container.image", "kubernetes.container.name", "kubernetes.labels",
+            "kubernetes.namespace", "kubernetes.node.name", "kubernetes.pod.name", "kubernetes.pod.uid",
+            "parent.id",
+            "processor.event", "processor.name",
+            "sourcemap.bundle_filepath", "sourcemap.service.name", "sourcemap.service.version",
+            "span.duration.us", "span.hex_id", "span.id", "span.name", "span.parent", "span.start.us", "span.sync",
+            "span.action", "span.subtype", "span.type",
+            "system.cpu.total.norm.pct", "system.memory.actual.free", "system.memory.total",
+            "system.process.cpu.total.norm.pct", "system.process.memory.rss.bytes", "system.process.memory.size",
+            "tags",
+            "timestamp.us",
+            "trace.id",
+            "transaction.duration.us", "transaction.id", "transaction.marks.navigationTiming", "transaction.name",
+            "transaction.result", "transaction.sampled", "transaction.span_count.dropped.total", "transaction.type",
+            "url.port",  # field copy from context.request.url.port
+            "url.scheme",  # field copy from context.request.url.protocol
+
+            # scripted fields
+            "error id icon",
+            "view errors",
+            "view spans",
+        }
+
+        # TBD
+        exception_fields.update({
+            "beat.timezone", "context.http.status_code", "context.response.finished",
```

Comment:
I think `beat.timezone` should be inherited from beats, so no need for a specific mapping on our side.

Reply:
Agreed!  I expect we'll sync with beats and be able to clear these exceptions before this gets merged.